yt_dlp/extractor/animeonegai.py

```diff
@@ -0,0 +1,103 @@
+            'tags': ['Romance', 'Comedia', 'Drama', 'Serie'],
+            'series': 'GTO',
+            'episode_number': 1,
+        },
+        'skip': 'Login Required',
+    }]
+    _DEVICE_ID = str(uuid.uuid4())
+    _HEADERS = {
+        'Authorization': None,
+        'Content-Type': 'application/json',
+        'Host': 'api.animeonegai.com',
+    }
+
+    def _perform_login(self, username, password):
+        if (username.lower() == 'token' and password is not None):
+            self._HEADERS['Authorization'] = f'Bearer {password}'
+            self.report_login()
+        elif (username.lower() != 'token' and password is not None):
+            raise ExtractorError('Login type not supported', expected=True)
+
+    def _real_extract(self, url):
+        if (self._HEADERS['Authorization'] is None):
+            raise ExtractorError(
+                'Use "--username token" and "--password <ott_token>" to log in. The ott_token can be found in Local Storage using the Developer Tools, after login in a browser window to Anime Onegai site', expected=True)
+
+        language, video_id = self._match_valid_url(url).group('language', 'id')
+
+        self._initialize_geo_bypass({
+            'countries': self._GEO_COUNTRIES,
+        })
+
+        chapter_data = self._download_json(
+            f'https://api.animeonegai.com/v1/restricted/chapter/entry/{video_id}',
+            video_id, 'Downloading Chapter Data', headers=self._HEADERS, query={'lang': language})
+
+        if (chapter_data.get('ID') == 0):
+            raise ExtractorError('Chapter data not found', expected=True)
+
+        asset_id = int_or_none(chapter_data.get('asset_id'))
+        asset_data = self._download_json(
+            f'https://api.animeonegai.com/v1/restricted/asset/public/{asset_id}',
+            video_id, 'Downloading Asset Data', headers=self._HEADERS, query={'lang': language, 'cache': 'true'})
+
+        video_entry = str_or_none(chapter_data.get('video_entry'))
+        video_data = self._download_json(
+            f'https://api.animeonegai.com/v1/media/{video_entry}',
+            video_id, 'Downloading Video Data', headers=self._HEADERS,
+            query={'tv': 'false', 'mobile': 'false', 'device_id': self._DEVICE_ID, 'platform': 'web'})
+
+        if (video_data.get('drm') and not self.get_param('allow_unplayable_formats')):
+            raise ExtractorError('The video is DRM protected', expected=True)
+
+        formats = self._extract_mpd_formats(video_data['dash'], video_id)
+        subtitles = {}
+
+        for subtitle in video_data.get('subtitles') or []:
+            subtitles.setdefault(str_or_none(subtitle.get('lang', language)), []).append({
+                'url': url_or_none(subtitle.get('url')),
+                'ext': determine_ext(str_or_none(subtitle.get('url'))),
+                'name': str_or_none(subtitle.get('name')),
+            })
```

Comment:
I can not test this, but I think this should work as well
```suggestion
        subtitles = traverse_obj(video_data, ('subtitles', ..., {
            'id': ('lang', {str}),
            'url': ('url', {url_or_none}),
            'name': ('name', {str}),
        }, all, {subs_list_to_dict(lang=language)}))
```